### PR TITLE
Review unreviewed code already in master

### PR DESCRIPTION
**Pull request by LukaszDygon**
This code in master needs to be reviewed

### CODE REVIEW

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/db/daos/RegionSummaryReportsDao.java [hunk 1]
```diff
@@ -27,16 +27,14 @@
...
-                    .executeAndReturnGeneratedKeys("report_id")
-                    .mapTo(Integer.class)
-                    .findOnly();
+            return handle.createScript("INSERT INTO region_summary_report (region_id, user_id, status, report_time, report_body)" +
+                    " VALUES ("
+                    + report.getRegionId() + ", "
+                    + report.getUserId() + ", "
+                    + report.getStatus() + ", '"
+                    + report.getReportTime() + "', '"
+                    + report.getReportBody()
```
Comment (Joe-Edwards) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/db/daos/RegionSummaryReportsDao.java]: Is this a deliberate SQL injection flaw?

File: AgentDiscoveries-Frontend/app/src/components/message.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { Alert } from "react-bootstrap"
+
+export class Message extends React.Component {
+
+    render() {
+        if (this.props.message.message !== "") {
+
+            return (
+                <Alert bsStyle={this.props.message.type} className={`${this.props.message.type}-message`}>
+                    {this.props.message ? this.props.message.message : ""}
```
Comment (Joe-Edwards) [inline AgentDiscoveries-Frontend/app/src/components/message.js]: Surely the message can never be falsey if it has reached this branch?

File: AgentDiscoveries-Frontend/app/src/components/search-forms/search-utilities.js [hunk 3]
```diff
@@ -30,7 +30,12 @@ export function getResultsAsynch(apiAddress, searchForm) {
 
     var response = searchAPI(apiAddress, searchParams)
         .then(response => {
-            return response.json();
+            if (response.status === 200) {
+                let results = response.json()
+                return results;
```
Comment (Joe-Edwards) [inline AgentDiscoveries-Frontend/app/src/components/search-forms/search-utilities.js]: `return response.json();`?

File: AgentDiscoveries-Frontend/app/src/components/submit-forms/agent-form.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { handleSubmit } from "./submit-utilities"
+import { Message } from "../message"
+
+export class CreateAgent extends React.Component {
+
+    constructor(props) {
+        super();
+        this.state = {
+            "submitForm": props.submitForm,
+            "onSubmit": props.onSubmit
```
Comment (Joe-Edwards) [inline AgentDiscoveries-Frontend/app/src/components/submit-forms/agent-form.js]: unnecessary quotes?

File: AgentDiscoveries-Frontend/app/src/components/submit-forms/agent-form.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    render(props) {
+        return (
+            <div className="col-md-12">
+                <Form onSubmit={this.state.onSubmit}>
+                    <h3>Create Agent</h3>
+
+                    <FormGroup>
+                        <ControlLabel>First Name</ControlLabel>
+                        <FormControl type="text" required
+                            inputRef={firstName => this.state.submitForm.firstName = firstName}
```
Comment (Joe-Edwards) [inline AgentDiscoveries-Frontend/app/src/components/submit-forms/agent-form.js]: I'm pretty sure you should never modify state directly in react. See https://reactjs.org/docs/forms.html

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/AgentDiscoveriesApplication.java [hunk 6]
```diff
@@ -102,12 +102,14 @@ private void agentsRouteGroup() {
         get("/:id", (req, res) -> agentsRoutes.readAgent(req, res, idParamAsInt(req)), responseTransformer);
         put("/:id", (req, res) -> agentsRoutes.updateAgent(req, res, idParamAsInt(req)), responseTransformer);
         delete("/:id", (req, res) -> agentsRoutes.deleteAgent(req, res, idParamAsInt(req)), responseTransformer);
+        get("", (req, res) -> agentsRoutes.readAgents(req, res), responseTransformer);
```
Comment (Joe-Edwards) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/AgentDiscoveriesApplication.java]: Am I right in thinking pagination will be added by the students?

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/db/daos/RegionSummaryReportsDao.java (hunk 1 above)
Comment (LukaszDygon) [reply AgentDiscoveries-Backend/src/main/java/org/softwire/training/db/daos/RegionSummaryReportsDao.java]: This is actually from the model answers to the training questions. I apologise for the merge request including this. I am unaware of whether this is deliberate

**Pull request closed without merging**